### ISSUE TITLE
messagesCollectionView has incorrect bottom inset when inside smaller container view

### ISSUE THREAD

**Issue by davidvpe** [MessageKit/MessageKit#761]
### General Information

Created a ChatViewController that inherits from MessagesViewController and add it's view to smaller view inside a parent view controller, so the Chat View is actually smaller than the device. This makes the bottom content inset act weird and doesn't let me see any new message that arrives or even my own last message.

- MessageKit Version:
1.0.0

- iOS Version(s):
11.4

- Swift Version:
4

- Devices/Simulators:
All simulator devices (my own iPhone 7 does it too)

- Reproducible in ChatExample? (Yes/No):
No, because in order to reproduce it you have to present the Chat View Controller as a smaller view

### What happened?

![ezgif-5-f92089b9c5](https://user-images.githubusercontent.com/1934159/42723239-92b8bdbc-875a-11e8-8a5f-de710235255b.gif)

### What did you expect to happen?

I expect the bottom content inset to be exactly the size of the keyboard not more not less.

### My workaround

Right now, I'm altering the handleKeyboardDidChangeState function, instead of this line:

`let newBottomInset = view.frame.height - keyboardEndFrame.minY - iPhoneXBottomInset`

I am doing this:

`let convertedKeyboardEndFrame = messagesCollectionView.convert(keyboardEndFrame, from: nil)`
`let intersect = convertedKeyboardEndFrame.intersection(messagesCollectionView.bounds)`
`let newBottomInset = intersect.size.height`

**Comment by SD10** [comment]
I'm going to label this a bug because it seems like the changes we made to the keyboard have caused a regression in container view controllers. Maybe we need to add one to the example app to check @nathantannar4?

**Comment by nathantannar4** [comment]
@SD10 good point, I'll work it into the new example app

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by nathantannar4** [comment]
Hmm, @davidvpe Im building a new example app and when I create a custom container the bottom inset is correct. I am doing:

```
final class ConversationContainerController: UIViewController {
    
    let conversationViewController = ConversationViewController()
    
    let bannerView = UIView()
    
    override var canBecomeFirstResponder: Bool {
        return conversationViewController.canBecomeFirstResponder
    }
    
    override var inputAccessoryView: UIView? {
        return conversationViewController.inputAccessoryView
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        conversationViewController.willMove(toParentViewController: self)
        self.addChildViewController(conversationViewController)
        view.addSubview(conversationViewController.view)
        conversationViewController.didMove(toParentViewController: self)
        view.addSubview(bannerView)
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        bannerView.frame = CGRect(origin: .zero, size: CGSize(width: view.bounds.width, height: 200))
        conversationViewController.view.frame = CGRect(x: 0, y: 200, width: view.bounds.width, height: view.bounds.height - 200)
    }
    
}
```

**Comment by nathantannar4** [comment]
hmm actually I do when the `navigationController?.navigationBar.isTranslucent = true`

*EDIT*
Scratch that it was something else, bottom content inset still correct

**Comment by SD10** [comment]
I think MessageKit sets `extendedLayoutIncludesOpaqueBars` true by default @nathantannar4 🤔

**Comment by nathantannar4** [comment]
![simulator screen shot - iphone x - 2018-07-14 at 19 43 35](https://user-images.githubusercontent.com/15272998/42729962-39d7e206-879e-11e8-8685-14fc325389d9.png)

**Comment by nathantannar4** [comment]
Ok, after further debugging, the bottomInset is correct at first, but as soon as the keyboard becomes visible it is incorrect. This means that WE ARE calculating the bottom inset incorrectly and I can confirm this is a bug

**Comment by SD10** [comment]
@nathantannar4 This bug was caused by the change in #707

**Comment by SD10** [comment]
Set `extendedLayoutIncludesOpaqueBars = false`

**Comment by davidvpe** [comment]
@SD10 I added the `extendedLayoutIncludesOpaqueBars = false` as you mentioned but It doesn't work.

**Comment by SD10** [comment]
Ok @davidvpe I'll keep looking for you. It's pretty difficult to support everyone's UI requirements. It's working for my app in production though 😢

**Comment by davidvpe** [comment]
I understand thanks for that.

**Comment by ljs19923** [comment]
Do you have news ? I have same problem.
Me thats works when the keyboard appear ect, but when i present a viewcontroller with self.presentVC, and then this VC disappear, i got the problem of display :/

**Comment by andreyvit** [comment]
@davidvpe @JulienLevallois Can you guys please check if PR #787 helps with this issue?

**Comment by SD10** [comment]
Should be fixed in #787
